### PR TITLE
Refactoring (API&Client)

### PR DESCRIPTION
**Pull request by NkFab**
### What will this PR do?

- Add react query
- Revamp axios interceptors
- Add request file for all the axios requests
- Handle errors and success responses
- Add the session handling in the server 
- Fix the authorization backslash issue (remove the backslash in the env file )
- Add react router dom v6
- Add linting scripts and fix for the server and client.

### Branch

feature/session-expiry-logout

### CODE REVIEW

File: api/.jshintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (kizule) [inline api/.jshintrc:1]: Shouldn't file be named .eslintrc?

Comment (kizule) [reply api/.jshintrc:1]: Or .eslintrc.json. I'm only sure that file shouldn't be named .jshintrc, as we don't use it.

Comment (kizule) [reply api/.jshintrc:1]: My mistake. I was confused.

**Comment by kizule** [comment]
When I click a button to sign in with the Google, this happens:

```
Error: Unknown authentication strategy "google"
    at attempt (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\passport\lib\middleware\authenticate.js:190:39)
    at authenticate (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\passport\lib\middleware\authenticate.js:367:7)
    at Layer.handle [as handle_request] (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\layer.js:95:5)
    at C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (C:\Users\zoran\Desktop\development\zotzen-web\api\node_modules\express\lib\router\index.js:341:12)
```

File: api/.jshintrc (hunk 1 above)
Comment (NkFab) [reply api/.jshintrc:1]: Glad you figured it out. The .jshint  file is complementing the linting errors.

**Comment by NkFab** [comment]
I'm working on that authentication issue.

**Pull request merged**